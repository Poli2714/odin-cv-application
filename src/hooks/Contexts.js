import { createContext } from 'react';

const IndexContext = createContext({
  activeIndex: null,
  setActiveIndex: () => {},
});

const KeyContext = createContext({
  activeKey: {
    eduActiveKey: null,
    expActiveKey: null,
  },
  setActiveKey: () => {},
});

const EduErrMsgContext = createContext({
  showEduErrMessage: false,
  setShowEduErrMessage: () => {},
});

const ExpErrMsgContext = createContext({
  showExpErrMessage: false,
  setShowExpErrMessage: () => {},
});

const PersonalInfoContext = createContext({
  personalInfo: {
    profilePicture: '',
    fullName: '',
    currentJobTitle: '',
    email: '',
    mobile: '',
    city: '',
  },
  setPersonalInfo: () => {},
});

const SummaryContext = createContext({
  summary: '',
  setSummary: () => {},
});

const ExperienceContext = createContext({
  experiences: {
    id: {
      jobTitle: '',
      employer: '',
      jobStartDate: '',
      jobEndDate: '',
      jobDescription: '',
    },
  },
  setExperiences: () => {},
});

const EducationContext = createContext({
  educations: {
    id: {
      degree: '',
      university: '',
      eduStartDate: '',
      eduEndDate: '',
      eduInfo: '',
    },
  },
  setEducations: () => {},
});

export {
  EducationContext,
  EduErrMsgContext,
  ExperienceContext,
  ExpErrMsgContext,
  IndexContext,
  KeyContext,
  PersonalInfoContext,
  SummaryContext,
};
