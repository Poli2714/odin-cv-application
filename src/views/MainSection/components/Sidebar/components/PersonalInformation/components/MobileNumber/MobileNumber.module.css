.telInput {
  composes: textField from 'src/components/forms/TextField/TextField.module.css';

  &:not(:focus, :placeholder-shown) + .invalid {
    position: relative;

    &::after {
      color: var(--clr-warning);
      content: '⚠️ Not a valid number (format: +12345678910)';
      font-size: 0.85rem;

      position: absolute;
      bottom: -1.25rem;
      left: 0;
    }
  }
}
