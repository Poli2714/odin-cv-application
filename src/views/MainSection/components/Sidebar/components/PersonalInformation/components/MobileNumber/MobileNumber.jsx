import { useContext } from 'react';

import { InputWrapper } from 'src/components/layout';
import { Label } from 'src/components/forms';
import { PersonalInfoContext } from 'src/hooks/Contexts';
import { isValidMobile } from 'src/utils';
import styles from './MobileNumber.module.css';

function MobileNumber() {
  const { personalInfo, setPersonalInfo } = useContext(PersonalInfoContext);
  const { mobile } = personalInfo;

  const handleChange = e =>
    setPersonalInfo({ ...personalInfo, mobile: e.target.value.trim() });

  return (
    <InputWrapper>
      <Label labelName="Mobile number" labelFor="mobile" />
      <input
        className={styles.telInput}
        id="mobile"
        name="mobile"
        onChange={handleChange}
        pattern="^\+[0-9]{11,}"
        placeholder="+12345678910"
        type="tel"
        value={mobile}
      />
      {isValidMobile(mobile) || (
        <div data-testid="warning" className={styles.invalid}></div>
      )}
    </InputWrapper>
  );
}

export default MobileNumber;
