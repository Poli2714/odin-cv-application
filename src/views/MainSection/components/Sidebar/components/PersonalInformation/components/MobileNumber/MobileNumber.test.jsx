import { expect, test, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useState } from 'react';

import MobileNumber from './MobileNumber';
import { PersonalInfoContext } from 'src/hooks/Contexts';

vi.mock('src/components/layout', () => ({
  InputWrapper: vi.fn(({ children }) => (
    <div data-testid="input-wrapper">{children}</div>
  )),
}));

vi.mock('src/components/forms', () => ({
  Label: vi.fn(() => <label htmlFor="mobile">Test label</label>),
}));

function MockParentComponent() {
  const [personalInfo, setPersonalInfo] = useState({ mobile: '' });

  return (
    <PersonalInfoContext.Provider value={{ personalInfo, setPersonalInfo }}>
      <MobileNumber />
    </PersonalInfoContext.Provider>
  );
}

test('renders MobileNumber', () => {
  render(<MobileNumber />);

  expect(screen.getByTestId('input-wrapper')).toMatchSnapshot();
  expect(screen.queryByTestId('warning')).not.toBeInTheDocument();
});

test('updates mobile value as a user types', async () => {
  const user = userEvent.setup();

  render(<MockParentComponent />);
  const mobileInput = screen.getByLabelText('Test label');

  expect(mobileInput).toHaveValue('');
  expect(screen.queryByTestId('warning')).not.toBeInTheDocument();

  await user.type(mobileInput, '123');
  expect(mobileInput).toHaveValue('123');
  expect(screen.queryByTestId('warning')).toBeInTheDocument();

  await user.clear(mobileInput);
  await user.type(mobileInput, '+12345678901');
  expect(mobileInput).toHaveValue('+12345678901');
  expect(screen.queryByTestId('warning')).not.toBeInTheDocument();
});
