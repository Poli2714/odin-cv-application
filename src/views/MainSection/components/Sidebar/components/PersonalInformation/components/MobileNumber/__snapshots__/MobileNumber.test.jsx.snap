// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders MobileNumber 1`] = `
<div
  data-testid="input-wrapper"
>
  <label
    for="mobile"
  >
    Test label
  </label>
  <input
    class="_telInput_1536b4"
    id="mobile"
    name="mobile"
    pattern="^\\+[0-9]{11,}"
    placeholder="+12345678910"
    type="tel"
    value=""
  />
</div>
`;
